ui/src/App.js

```diff
@@ -133,22 +136,45 @@ class App extends Component {
     datasetApi.datasetGet(datasetCallback);
   }
 
+  getFacetMap(facets) {
+    var facetMap = new Map();
+    facets.forEach(function(facet) {
+      facetMap.set(facet.es_field_name, facet);
+    });
+    return facetMap;
+  }
+
   handleSearch(searchResults) {
+    // If a facet is selected from the dropdown, searchResults is the selected facet.
+    // If a facetValue is selected from the dropdown, searchResults is an Array
+    // of all selected facet values.
     let extraFacetEsFieldNames = this.state.extraFacetEsFieldNames;
+    // An array of selected facet values, of the form esFieldName=facetValue.
+    let newFilters = [];
     for (let i = 0; i < searchResults.length; i++) {
-      if (searchResults[i].facetValue == "") {
-        extraFacetEsFieldNames.push(searchResults[i].esFieldName);
+      if (searchResults[i].facetValue != "") {
+        newFilters.push(
+          searchResults[i].esFieldName + "=" + searchResults[i].facetValue
+        );
       }
+      extraFacetEsFieldNames.push(searchResults[i].esFieldName);
     }
+    // An array of existing selected facet values, of the form esFieldName=facetValue.
+    let currentFilters = this.filterMapToArray(this.state.filterMap);
```

Comment:
Can we get rid of all this filter logic for now, since we're not showing any facet values in the drop-down yet? It's hard to see if the code is right, if there's no way to exercise it.

Reply:
This is very much used now. We add and remove filters by checking and unchecking boxes.